src/main/java/hudson/maven/MavenBuild.java

```diff
@@ -837,9 +837,13 @@ protected Result doRun(BuildListener listener) throws Exception {
                         LOGGER.fine( "using maven 3 " + mavenVersion );
                         factory = new Maven32ProcessFactory( getParent().getParent(), MavenBuild.this, launcher, envVars, getMavenOpts(listener, envVars), null );
                         break;
-                    default:
+                    case MAVEN_3_3:
                         LOGGER.fine( "using maven 3 " + mavenVersion );
                         factory = new Maven33ProcessFactory( getParent().getParent(), MavenBuild.this, launcher, envVars, getMavenOpts(listener, envVars), null );
+                        break;
+                    default:
```

Comment:
I do not like such design, because it is "I am feeling lucky" when anybody tries a newer version of Maven and gets issue like JENKINS-43446 . Maybe it is better to just handle `case MAVEN_3_5:` and fail otherwise